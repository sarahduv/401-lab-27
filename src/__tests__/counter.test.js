import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Counter from '../components/counter';

Enzyme.configure({ adapter: new Adapter() });

describe('Testing our Counter component', () => {
  it('Counter changes state on Click', () => {
    let app = mount(<Counter />);
    let button = app.find('button');
    expect(button).toBeDefined();
    expect(app.state('count')).toBe(0);
    button.simulate('click');
    expect(app.state('count')).toBe(1);
  });

  it('render correctly', () => {
    const renderTree = renderer.create(<Counter />).toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
