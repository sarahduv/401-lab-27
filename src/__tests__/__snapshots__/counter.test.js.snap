// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing our Counter component render correctly to the DOM 1`] = `
<section
  className="counter"
>
  <button
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="count "
  >
    0
  </span>
  <button
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </button>
</section>
`;
